### ISSUE TITLE
linter-eslint doesn't respect .eslintignore 

### ISSUE THREAD

**Issue by oriSomething** [AtomLinter/linter-eslint#279]
`linter-eslint` version `5.1.0`
`atom` version `1.2.0`
`OS X` Yosemite

`linter-eslint` not ignoring folders in `.eslintignore`

when i try `eslint` from CLI:
```
/···/···.js
  0:0  warning  File ignored because of your .eslintignore file. Use --no-ignore to override
```

the `eslint` error for the file that should be ignored is "unused var"

**Comment by steelbrain** [comment]
I missed this while rewriting this linter, Thanks for the report! I'll be implementing it soon but would appreciate a PR :)

**Comment by oriSomething** [comment]
i can try writing a PR for this... see if i manage well :)

**Comment by oriSomething** [comment]
@steelbrain i run into issue when i add `--ignore-path` to `process.argv` it ignore the `.eslintignore` but if i put another path it throw an error of not finding `.eslintignore`. do you have a clue where it comes from? is there any special in atom environment

(worker.js)

**Comment by steelbrain** [comment]
@oriSomething if I had to guess, it's because we are linting a file instead of a dir, I mean `eslint lib` could ignore files mentioned in `.eslintignore`, but `eslint lib/some-ignored-file.js` wouldn't.

You can work it around by keeping a file system cache of the `.eslintignore` file and then using `minimatch` module to match each rule against the current file path

**Comment by IanVS** [comment]
If you tell ESLint to lint a file which is supposed to be ignored, it will normally give you a warning.  I don' know how this operates when you're passing in text with `--stdin`, however.

**Comment by oriSomething** [comment]
@steelbrain  yep... it seems like a problem with `eslint` itself. if you give absolute path even if it's not a specific file it will ignore `.eslintignore`. i'll see if there is an known issue at `eslint` repo

**Comment by IanVS** [comment]
Unless I'm missing it, it doesn't look like you're giving ESLint the filename.  You need to use `--stdin-filename <relative filename>`

**Comment by steelbrain** [comment]
@IanVS Instead of using that parameter, we're just passing it as the last parameter, which works too

**Comment by IanVS** [comment]
Hmmm, no those are different things.  When I use `--stdin-filename`, ESLint correctly ignores the file.

**Comment by IanVS** [comment]
Passing the file as the last parameter is how you would normally specify glob paths to lint, when you are not using `--stdin`.  Those options are not intended to be used together.

**Comment by steelbrain** [comment]
@IanVS Good catch! Thanks!

**Comment by oriSomething** [comment]
ok... i'm just confuse

**Comment by IanVS** [comment]
I think you just need `--stdin-filename` on the line before: https://github.com/AtomLinter/linter-eslint/blob/master/lib/worker.js#L92

**Comment by oriSomething** [comment]
noop... still doesn't ignore

**Comment by oriSomething** [comment]
we can work around this with parsing `.eslintignore` but i don't thing it's a good idea.
the file that contains the ignoring process is `eslint/lib/ignored-paths.js`

**Comment by steelbrain** [comment]
@oriSomething I have an idea, I'll give this a try

**Comment by micky2be** [comment]
Problem still exist with linter-eslint 5.2.6

**Comment by IanVS** [comment]
@micky2be Please open a new issue with all relevant details (a reproducible config/repo is best).
